Develop

gtortajada opened the pull request:
Coisas que ainda faltam polir: 
*Acertar espçamento lateral do conteúdo de todo o site.
*Mensagem padrão encaminhando para o whats deles.

* Footer
	Desktop: a lgoo deve ficar na esquerda e o iframe na direita.
*Testimonials:
	Video:
		Clicar para ativar som do vídeo
		Clicar novamente para tirar o som do vídeo
	A altura ocupada pelos cards deve ser a mesma do vídeo.

gtortajada committed:
build: Criação de sections vazias.

gtortajada committed:
feat: Criação de Footer.

gtortajada committed:
feat: Criação de navbar genérica.

gtortajada committed:
style: Adição de favicon na branch navbarAndFooter.

gtortajada committed:
Merge branch 'estrutura-site' into develop

gtortajada committed:
Merge branch 'navbarAndFooter' into develop

gtortajada committed:
fix: exclusão de arquivos duplicados.

gtortajada committed:
fix: exclusão de arquivos duplicados.

gtortajada committed:
style: Adição de fonte no projeto + adição no global.css.

gtortajada committed:
feat: Criação de texto básico nas sections. + Correção de nome AboutTs.

gtortajada committed:
feat: Criação de navbar e footer padrão com logo.

gtortajada committed:
style: Adição  + correção de fonte. *Existe um erro na fonte sem usar bold.

gtortajada committed:
Merge pull request #4 from gtortajada/feature/navbarAndFooter

Feature/navbar and footer

gtortajada committed:
style: Adição de todos os assets no projeto.

gtortajada committed:
debug: layout funcional + erro no 'myTheme" que não quebra, mas impede de usar propriedades de breackpoints.

gtortajada committed:
fix: Alteração de nome de arquivo + exclusão do .css footer.

gtortajada committed:
chore: update dev dependencies + refactoring theme config + add .vscode .gitgnore

gtortajada committed:
feat: footer v1.

gtortajada committed:
feat: faq v1

gtortajada committed:
feat: Testimonials v1

gtortajada committed:
feat: FAQ v1.

gtortajada committed:
style: Correção botão cta.

gtortajada committed:
feat: Criação de componente botão cta v1.

gtortajada committed:
feat: Criação de aboutUs v1 + correção de exportação faq. *Antes de testar a marquee.

gtortajada committed:
feat: aboutUs v2 + adição de Marquee para logos girando infinitamente.

gtortajada committed:
feat: services v1 + update versão mantine/core e instalação do carousel mantine (embla-carousel-autoplay).

gtortajada committed:
feat: navbar v1

gtortajada committed:
style: correção de padding e cor.

gtortajada committed:
feat: hero v1.

gtortajada committed:
feat: hero v2. Correções de styles.

gtortajada committed:
style: Ajuste de paddings.

gtortajada committed:
style: Correção de cor de bg do site.

gtortajada committed:
style: Correção de espaçamentos.

gtortajada committed:
style: Correção de espaçamentos.

gtortajada committed:
style: Correção de espaçamentos.

gtortajada committed:
style: Correção de cores.

gtortajada committed:
style: Ajuste simples.

gtortajada committed:
chore(env): Especifica a versão do Node.js do projeto.

gtortajada committed:
chore: Adição de config de images.

gtortajada committed:
feat: Estilização e correção de navbar.

gtortajada committed:
feat: Estilização e correção de funcionalidade footer.

gtortajada committed:
style: Estilização e finalização de componentes.
*Falta finalizar vídeo do testimonials.

the-almeida commented on next.config.ts:
```diff
@@ -1,7 +1,11 @@
-  /* config options here */
-};
+  images: {
+    dangerouslyAllowSVG: true,
```
@gtortajada pq vc precisa que o next otimize os seus SVGs?

the-almeida replied:
low prio

gtortajada replied:
Por ser SVG não sei se precisa tanto como se fosse png por exemplo, mas pelo next ser bem competente  preferi ele cuidando disso.

the-almeida commented on package.json:
```diff
@@ -9,12 +9,16 @@
     "react": "^19.0.0",
-    "react-dom": "^19.0.0"
+    "react-dom": "^19.0.0",
+    "react-fast-marquee": "^1.6.5"
```
@gtortajada vc está usando tanto o embla quanto o fast-marquee?

the-almeida replied:
low prio

the-almeida commented on public/videos/videoExperienciasReais.MOV:
@gtortajada subir binários grandes (tipo esse vídeo) não é uma prática comum. Tbm não é um problema, mas é importante vc saber que normalmente videos e binários mais pesados não ficam no git

gtortajada replied:
Eu vi que isso era um problema, mas dei prioridade para o vídeo em si.

the-almeida commented on src/components/Footer.tsx:
```diff
@@ -0,0 +1,223 @@
+    { link: "#faq", label: "Perguntas frequentes" },
+  ];
+
+  const handleLinkClick = (
```
@gtortajada Essa lógica não deveria ser global? Como esse click atualiza o highlight da navbar?

the-almeida replied:
Comecei a investigar e vi que vc tem uma função muito parecida na Navbar. Além disso existe um bug real que é a Navbar não identificar o scroll pela página (pra mudar o item que está ativo)

the-almeida replied:
Como um exercício pra vc evoluir eu recomendo vc usar a ContextAPI, e criar um CustomHook pra observar o scroll, mas tem jeitos mais simples de resolver (mas a caminho dos preguiçosos eu só conto dps que vcs tiver resolvido o problema sozinho kkk)

the-almeida replied:
high prio @gtortajada created a scrappy solution and will finish applying it today

the-almeida replied:
Solved on ab4411badaa5b6f21445f9f17fe3ade909e9cf19

the-almeida commented on src/components/Navbar.tsx:
```diff
@@ -0,0 +1,129 @@
+  };
+
+  const items = links.map((link) => {
+    const handleLinkClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
```
@gtortajada A lógica dessa função está duplicada no footer e o estado do item ativo (active/setActive) não leva em consideração o scroll da página. 

Já iniciei essa discussão no [comentário que fiz no footer](https://github.com/gtortajada/sisconcontabilidade.com/pull/5/files#r2317668202), então vamos manter centralizado lá.

gtortajada replied:
Corrigir: Adicionar onClick na logo com scroll para a home.

the-almeida replied:
Fiz o esquema do scroll top (commit 5a743cafd7df1ef776da9e3e0eb8b8ac9e2906d7) e dei uma refatorada na lógica de criar os itens da navbar. Vc pode usar essa refatoração de referência pra melhorar o código que a IA gerou pra vc na solução do scrollSpy, foca principalmente na estrutura e nomenclatura das funções/variáveis.

the-almeida commented on src/components/Footer.tsx:
Logo do siscon na versão desktop precisa de margin no eixo Y, pode ser 8px (ou qqr numero/unidade que vc tenha como padrão no projeto pra margem pequena)

the-almeida replied:
low prio

the-almeida replied:
Solved on 87fa3376d3977650da315261aa283e9725b04731

the-almeida commented on src/components/Navbar.tsx:
```diff
@@ -0,0 +1,129 @@
+
+  const headerStyles = {
+    backgroundColor: "#0161DF",
+    position: "fixed" as const,
```
@gtortajada nunca vi essa sintaxe "as const" que vc usou, isso faz o que?

the-almeida replied:
low prio

the-almeida replied:
<img width="1123" height="674" alt="image" src="https://github.com/user-attachments/assets/49051302-f5c2-4c1f-8ad0-7906ce4d9a99" />

@gtortajada Não parece uma boa prática usar isso sem contexto específico kkk mas não gastei muito tempo testando ou refletindo kkk não parece causar bugs por agora, mas talvez eu experimente tirar pra ver se interfere em algo

the-almeida replied:
@gtortajada  Removi o "as const" e resolvi o erro original com a tipagem correta pros objetos de estilos (commit 0bd56f50437839463a9e6daf1a27b0b81bf35bb1). Também vou colocar aqui um explicação de IA sobre o problema, solução e uma possivel opção para soluções diferentes no futuro: 
<img width="896" height="436" alt="image" src="https://github.com/user-attachments/assets/b469ef3b-0436-42e7-8a9d-05436d06fe34" />

the-almeida commented on src/components/sections/aboutUs/aboutUs.tsx:
```diff
@@ -0,0 +1,376 @@
+  { name: "Akon", src: "/images/logoClientes/akon.png" },
+];
+
+const logosForMarquee = [...logos, ...logos, ...logos, ...logos];
```
@gtortajada  Meio estranho isso aqui kkkkk eu lembro do contexto que o loop não tava funcionando direito, mas vale por um comentário explicando o pq dessa "hacky solution" (ou então resolver a raiz do problema)

the-almeida replied:
medium prio

the-almeida replied:
solved on 9ef609c9abfde0ad14f0ce1a07bfb9934bc8609b

the-almeida commented on src/components/sections/testimonials/testimonials.tsx:
```diff
@@ -0,0 +1,189 @@
+            </AspectRatio>
+          </Stack>
+
+          <CtaButton>Fale com nossos especialistas agora</CtaButton>
```
@gtortajada remove o agora, esse CTA ta mto grande, vai estourar tela em mobile

the-almeida replied:
low prio

the-almeida replied:
@gtortajada removi o agora no commit 926522c2e477259acab60a85622a241a7756c3e9

the-almeida commented on src/components/ctaButton.tsx:
@gtortajada esse botão não funciona kkk tem o efeito visual do clique, mas clicar não redireciona pro whatsapp

the-almeida replied:
high high prio

the-almeida commented on src/components/sections/aboutUs/aboutUs.tsx:
```diff
@@ -0,0 +1,376 @@
+                  zIndex: 1,
+                }}
+              />
+              <Box style={{ position: "relative", zIndex: 2 }}>
```
@gtortajada Esse card tá sem margens laterais no mobile, talvez esteja estourando layout por conta do CTA grande, ou realmente só precise de um margin left e right de 4px

the-almeida replied:
low prio

the-almeida replied:
@gtortajada I tried a fix that worked on my main usecase (commit 72692fd336dc1f5f293367684c18506753a9936b) but didn't stress out testing on different devices.

the-almeida commented on src/components/sections/hero/hero.tsx:
```diff
@@ -0,0 +1,209 @@
+        pb={{ base: 0, md: rem(150) }}
+      >
+        <Stack align="center" style={{ zIndex: 1 }} p="sm" gap="xs">
+          <Title
```
@gtortajada no mobile falta margem entre esse titulo e a navbar, e se for um celular no modo paisagem o titulo fica sobreposto e nem aparece (então o problema provavelmente é o body inteiro não levar em consideração a navbar, isso tbm deve causar problema no scroll após clicar em um item)

the-almeida replied:
high prio

gtortajada replied:
Percebi também que ter uma altura de hero definida iria quebrar o layout caso o dispositivo do usuário tenha alguma função de aumento de fonte (alteração de acessibilidade), com isso não defini altura (só mínima). Alterei também para que o conteúdo da hero ocupe toda a tela (alteração estética).

Para resolver, adicionei paddingTop (referente ao tamanho da navbar) ao body, alterei o minHeight (base) para 100vh e no breakpoint md 768px para garantir responsividade.

the-almeida commented on src/components/sections/services/services.tsx:
```diff
@@ -0,0 +1,238 @@
+          {isMobile ? (
+            <Box w="100%">
+              <Carousel
+                withIndicators
```
@gtortajada os indicators poderiam ter uma margin ou padding top pra sair de cima dos items

the-almeida replied:
low prio

the-almeida replied:
@gtortajada this one is good for pair programming alongside the video fixes and improvements (dicussion here: https://github.com/gtortajada/sisconcontabilidade.com/pull/5#discussion_r2317784144)

the-almeida commented on src/components/sections/testimonials/testimonials.tsx:
```diff
@@ -0,0 +1,189 @@
+} from "@mantine/core";
+import Image from "next/image";
+
+const testimonials = [
```
@gtortajada precisa corrigir o conteudo dos testimonials usando de ref o que foi conversado com a Helo no grupo

the-almeida replied:
high high prio @the-almeida

the-almeida replied:
Solved on 2480e7308bf7a2d4302b60d23874ae9832c2ae7c

the-almeida commented on src/theme.ts:
```diff
@@ -0,0 +1,14 @@
+import { createTheme } from '@mantine/core';
+
+export const theme = createTheme({
+  fontFamily: 'Aeonik, sans-serif',
```
@gtortajada foi intencional não colocar fonte de fallback?

the-almeida replied:
low prio

the-almeida replied:
Solved on 0bd34d0d7b0859ff4f644c18b31528739f8e5fb1

the-almeida replied:
solved on ab4411badaa5b6f21445f9f17fe3ade909e9cf19

the-almeida commented on src/components/sections/aboutUs/aboutUs.tsx:
```diff
@@ -0,0 +1,376 @@
+                              é empreender no Brasil.
+                            </Text>
+                            <CtaButton>
+                              Tire suas dúvidas pelo WhatsApp
```
@gtortajada notei que você duplicou todo o conteúdo nessa section, tem um bom motivo?

the-almeida replied:
low low prio

the-almeida replied:
@gtortajada entendi o motivo de ter o mesmo conteúdo usado 2x no mesmo component (devido a estrutura da section e paralax somente no mobile), então resolvi isso extraindo o conteúdo para uma variável estruturada que pode ser reutilizada facilitando atualizações e manutenções futuras. Também aproveitei pra fazer algumas refatorações então é importante vc revisar as mudanças que fiz nesse arquivo. Commits abaixo: 
72692fd336dc1f5f293367684c18506753a9936b
9ef609c9abfde0ad14f0ce1a07bfb9934bc8609b

the-almeida commented on src/components/sections/testimonials/testimonials.tsx:
```diff
@@ -0,0 +1,189 @@
+                style={{ borderRadius: "8px" }}
+              >
+                <video
+                  poster="/images/video-poster.jpg"
```
@gtortajada essa img tá dando 404 no terminal (e acho que ela não existe mesmo)

the-almeida replied:
low prio because the video is being reproduced, we will pair program the solution of all problems related to video (no cover image, no video controls, no-lazy loading, starting before visible, always muted, is it acceptable to keep on git?)

the-almeida replied:
Discussion about not adding large binaries to git: https://github.com/gtortajada/sisconcontabilidade.com/pull/5#discussion_r2317660910

the-almeida committed:
fix: wrong syntax on inline css media query + lint

gtortajada committed:
feat: Adição de mensagem e função encaminhar ao botão de contato.

gtortajada committed:
fix: Correção de conteúdo conflitando com navbar.

-Adição de paddingTop ao body no globals.css.
-Exclusão de height 90-95vh na hero.
-Adição de minHeight 100vh na base e 768px no breakpoint md.

the-almeida committed:
fix: updated boilerplate testimonials

the-almeida committed:
Merge branch 'develop' of github.com:gtortajada/sisconcontabilidade.com into develop

the-almeida committed:
fix: wrong 'max-width' syntax on inline cssinjs

the-almeida committed:
refactor: remove unecessary comments + fix missing margin on mobile AboutUs card + alt image description + removed content duplication

the-almeida committed:
fix: removed client logos infinite scroll hacky solution + code clarity improvement

the-almeida committed:
fix: adding key props to highlight paragraphs

the-almeida committed:
refactor: shortened cta button text

the-almeida committed:
refactor: adding my on Footer logo

the-almeida committed:
refactor: removed unecessary verbosity ('as const') and typed correctly the style variables

the-almeida committed:
refactor: improved variable namming

the-almeida committed:
feat: made Navbar logo scroll to top + refactored navbarLinks creation

the-almeida committed:
refactor: added Arial as fallback font

the-almeida committed:
fix: scroll not updating navbar active item
